File: 30-days-of-code/php/day-26-nested-logic.php
<?php
$_fp = fopen("php://stdin", "r");
// Begin Solution
//---------------------------------------------------------------------------------------
list($dayIn, $monthIn, $yearIn) = array_map('intval', explode(' ', fgets(STDIN)));
list($dayOut, $monthOut, $yearOut) = array_map('intval', explode(' ', fgets(STDIN)));
$fineDay = 15;
$fineMonth = 500;
$fineYear = 10000;
if ($yearIn == $yearOut) {
    if ($monthIn == $monthOut) {
        if ($dayIn > $dayOut) {
            echo ($dayIn - $dayOut) * $fineDay;
        } else {
            echo 0;
        }
    } else if ($monthIn > $monthOut) {
        echo ($monthIn - $monthOut) * $fineMonth;
    } else {
        echo 0;
    }
} else if ($yearIn > $yearOut) {
    echo $fineYear;
} else {
    echo 0;
}
//---------------------------------------------------------------------------------------
// End Solution

?>

File: 30-days-of-code/cpp/day-26-nested-logic.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    // Begin Solution
    //---------------------------------------------------------------------------------------
    int dayIn, monthIn, yearIn, dayOut, monthOut, yearOut;
    cin >> dayIn >> monthIn >> yearIn;
    cin >> dayOut >> monthOut >> yearOut;
    int fineDay = 15;
    int fineMonth = 500;
    int fineYear = 10000;
    if (yearIn == yearOut) {
        if (monthIn == monthOut) {
            if (dayIn > dayOut) {
                cout << (dayIn - dayOut) * fineDay;
            } else {
                cout << 0;
            }
        } else if (monthIn > monthOut) {
            cout << (monthIn - monthOut) * fineMonth;
        } else {
            cout << 0;
        }
    } else if (yearIn > yearOut) {
        cout << fineYear;
    } else {
        cout << 0;
    }
    //---------------------------------------------------------------------------------------
    // End Solution  
    return 0;
}
